demos/python_demos/text_detection_demo/text_detection_demo.py

```diff
@@ -0,0 +1,260 @@
+               (6, x, y + 1), (7, x + 1, y + 1)]
+
+        return [i for i in tmp if i[1] >= 0 and i[1] < w and i[2] >= 0 and i[2] < h]
+
+    def _get_neighbours_4(self, x, y):
+        w, h = self.w, self.h
+        tmp = [(1, x, y - 1),
+               (3, x - 1, y),
+               (4, x + 1, y),
+               (6, x, y + 1)]
+
+        return [i for i in tmp if i[1] >= 0 and i[1] < w and i[2] >= 0 and i[2] < h]
+
+    def _mask_to_bboxes(self, min_area=300, min_height=10):
+        image_h, image_w = self.image_shape
+        self.bboxes = []
+        max_bbox_idx = self.mask.max()
+        mask_tmp = cv2.resize(self.mask, (image_w, image_h), interpolation=cv2.INTER_NEAREST)
+
+        for bbox_idx in range(1, max_bbox_idx + 1):
+            bbox_mask = mask_tmp == bbox_idx
+            cnts, _ = cv2.findContours(bbox_mask.astype(np.uint8), cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+            if len(cnts) == 0:
+                continue
+            cnt = cnts[0]
+            rect, w, h = self._min_area_rect(cnt)
+            if min(w, h) < min_height:
+                continue
+            if w * h < min_area:
+                continue
+            self.bboxes.append(self._order_points(rect))
+
+    def _min_area_rect(self, cnt):
+        rect = cv2.minAreaRect(cnt)
+        w, h = rect[1]
+        box = cv2.boxPoints(rect)
+        box = np.int0(box)
+        return box, w, h
+
+    def _order_points(self, rect):
+        """ (x, y)
+            Order: TL, TR, BR, BL
+        """
+        tmp = np.zeros_like(rect)
+        sums = rect.sum(axis=1)
+        tmp[0] = rect[np.argmin(sums)]
+        tmp[2] = rect[np.argmax(sums)]
+        diff = np.diff(rect, axis=1)
+        tmp[1] = rect[np.argmin(diff)]
+        tmp[3] = rect[np.argmax(diff)]
+        return tmp
+
+    def decode(self):
+        "Decode loaded data to ``self.bboxes``"
+        for point in self.points:
+            y, x = point
+            neighbours = self._get_neighbours(x, y)
+            for n_idx, nx, ny in neighbours:
+                link_value = self.link_mask[y, x, n_idx]
+                pixel_cls = self.pixel_mask[ny, nx]
+                if link_value and pixel_cls:
+                    self._join(point, (ny, nx))
+
+        self._get_all()
+        self._mask_to_bboxes()
+
+    def plot_result_cvgui(self, image):
+        "Plot and show decoded results via OpenCV's GUI"
+        for box in self.bboxes:
+            cv2.drawContours(image, [box], 0, (0, 0, 255), 2)
+        cv2.imshow('Detected text', image)
+        cv2.waitKey(0)
+        if cv2.waitKey():
+            cv2.destroyAllWindows()
+
+
+def main():
+    if args.model_path.endswith('.xml'):
+        td = cv2.dnn.readNet(args.model_path, args.model_path[:-3] + 'bin')
+    else:
+        print("Not valid model's XML file name (should be something like 'foo.xml')")
+        sys.exit()
+    img = cv2.imread(args.image_path)
+    blob = cv2.dnn.blobFromImage(img, 1, (1280, 768))
```

Comment:
You need to check that the model input is actually of that size first. Likewise, you need to check that the outputs are of the expected size before you start using them.

Reply:
`blobFromImage()` will always return an array (1280, 768) if img is 2d (HxW) or 3d array (HxWxC; where C == 3)

Reply:
The issue is not with `blobFromImage`, but with the model. The user might not necessarily pass you a model that takes input with these dimensions. Likewise for the output.

Reply:
Than it will crash

Reply:
Yes it will. That is the problem.

It should instead print a message saying that the model is incompatible with the demo.

Reply:
Could you give me a hint on how I could do that appropriately? Nothing comes to my mind except catching exceptions. I do not know how to get model input/output dimensions, not from docs that describe it.

Reply:
Admittedly, I have made the original comment with the expectation that you would have switched to the Inference Engine API by the time you addressed it. :-)

It does seem like OpenCV offers no way to examine the expected input size, so we'll have to give up on checking that. However, you could at least check the dimensions of `a` and `b` before you pass them to `dcd.load`.

Reply:
> However, you could at least check the dimensions of a and b before you pass them to dcd.load

I added this and check for `getUnconnectedOutLayersNames()` output.